### PR TITLE
V47 v48 v49 database changes

### PR DESCRIPTION
**Pull request by HMCTSDeveloper**
Implementing Barry's changes for some of v47, v48 and v49. These do not include transcription changes as they are being handled with the new transcription Jira's.

### CODE REVIEW

File: src/main/resources/db/migration/common/V1_219__v48_without_transcription.sql [hunk 1]
```diff
@@ -0,0 +1,17 @@
+-- V48 without transcription changes.
+ALTER TABLE court_case ALTER COLUMN case_number SET NOT NULL;
+ALTER TABLE court_case ALTER COLUMN case_closed SET NOT NULL;
+ALTER TABLE court_case ALTER COLUMN interpreter_used SET NOT NULL;
+
+ALTER TABLE node_register ALTER COLUMN device_type SET DEFAULT 'DAR';
+
+ALTER TABLE hearing ALTER COLUMN hearing_date SET NOT NULL;
+--ALTER TABLE hearing ALTER COLUMN scheduled_start_time SET NOT NULL;
```
Comment (HMCTSDeveloper) [inline src/main/resources/db/migration/common/V1_219__v48_without_transcription.sql:9]: not sure about this change , I think in some cases like darts events we don't have a scheduled start time.

File: src/main/resources/db/migration/common/V1_218__node_register_missing_fk_and_seq.sql [hunk 2]
```diff
@@ -0,0 +1,18 @@
+-- v47
+-- changes were not kept update to date for the node_register table so its being done now with the new
+-- changes for the table in v47.
+CREATE SEQUENCE node_seq START WITH 10000;
+
+ALTER TABLE node_register
```
Comment (davet1985) [inline src/main/resources/db/migration/common/V1_218__node_register_missing_fk_and_seq.sql:6]: I think this table should be called `device_register`?

File: src/main/resources/db/migration/common/V1_217__drop_courthouse_region_cra_id.sql [hunk 3]
```diff
@@ -0,0 +1,5 @@
+-- V47
+ALTER TABLE courthouse_region_ae DROP COLUMN cra_id;
+CREATE UNIQUE INDEX courthouse_region_ae_pk ON courthouse_region_ae(cth_id,reg_id);
+
+
```
Comment (davet1985) [inline src/main/resources/db/migration/common/V1_217__drop_courthouse_region_cra_id.sql:5]: Do we also need to drop the sequence?

```
DROP SEQUENCE IF EXISTS cra_seq;
```

File: src/main/resources/db/migration/common/V1_218__node_register_missing_fk_and_seq.sql [hunk 4]
```diff
@@ -0,0 +1,18 @@
+-- v47
+-- changes were not kept update to date for the node_register table so its being done now with the new
+-- changes for the table in v47.
+CREATE SEQUENCE node_seq START WITH 10000;
```
Comment (davet1985) [inline src/main/resources/db/migration/common/V1_218__node_register_missing_fk_and_seq.sql:4]: Barry's comment says 50000?

> amending der_seq to restart at 50000

Comment (davet1985) [inline src/main/resources/db/migration/common/V1_218__node_register_missing_fk_and_seq.sql:4]: Should this be called `der_seq`?

File: src/main/resources/db/migration/common/V1_218__node_register_missing_fk_and_seq.sql (hunk 2 above)
Comment (HMCTSDeveloper) [reply src/main/resources/db/migration/common/V1_218__node_register_missing_fk_and_seq.sql:6]: Talking to Barry he does not have a preference for this table and it seems like Sashi made a request for it to be node register. So I have decided to keep it as it was previously. Barry needs to change his script.

File: src/main/resources/db/migration/common/V1_219__v48_without_transcription.sql [hunk 5]
```diff
@@ -0,0 +1,17 @@
+-- V48 without transcription changes.
+ALTER TABLE court_case ALTER COLUMN case_number SET NOT NULL;
+ALTER TABLE court_case ALTER COLUMN case_closed SET NOT NULL;
+ALTER TABLE court_case ALTER COLUMN interpreter_used SET NOT NULL;
+
+ALTER TABLE node_register ALTER COLUMN device_type SET DEFAULT 'DAR';
```
Comment (davet1985) [inline src/main/resources/db/migration/common/V1_219__v48_without_transcription.sql:6]: Move this into 218?

File: src/main/resources/db/migration/common/V1_217__drop_courthouse_region_cra_id.sql (hunk 3 above)
Comment (HMCTSDeveloper) [reply src/main/resources/db/migration/common/V1_217__drop_courthouse_region_cra_id.sql:5]: yes its being done in V1_219__v48_without_transcription.sql

File: src/main/resources/db/migration/common/V1_219__v48_without_transcription.sql (hunk 1 above)
Comment (davet1985) [reply src/main/resources/db/migration/common/V1_219__v48_without_transcription.sql:9]: Yeah there's lot of nulls there

File: src/main/resources/db/migration/common/V1_217__drop_courthouse_region_cra_id.sql (hunk 3 above)
Comment (davet1985) [reply src/main/resources/db/migration/common/V1_217__drop_courthouse_region_cra_id.sql:5]: I've just seen this down below, maybe move it into here?

**Comment by barry-jones50410** [comment]
I think we discussed this on Teams, and my reply was "the legacy table is "tbl_moj_node" with a PK of node_id.
Sashi requested to create this table in modernised as "device_register" in expectation that other devices might be registered

File: src/main/resources/db/migration/common/V1_218__node_register_missing_fk_and_seq.sql (hunk 2 above)
Comment (richardbull) [reply src/main/resources/db/migration/common/V1_218__node_register_missing_fk_and_seq.sql:6]: Fingers crossed

File: src/main/resources/application-stg.yaml
Comment (richardbull) [inline src/main/resources/application-stg.yaml:1]: This looks to be a mistake - pretty sure we don't use this `stg` profile. Didn't @chrisbellingham-hmcts remove it recently from the README?

File: src/main/resources/db/migration/common/V1_213__dailylist_job_status_not_null.sql [hunk 6]
```diff
@@ -1 +1,2 @@
+--v48
```
Comment (richardbull) [inline src/main/resources/db/migration/common/V1_213__dailylist_job_status_not_null.sql:1]: Does adding this comment not invalidate the flyway schema history - checksum stuff?
Think it should be reverted

Comment (HMCTSDeveloper) [reply src/main/resources/db/migration/common/V1_213__dailylist_job_status_not_null.sql]: ACCEPT.

File: src/main/resources/application-stg.yaml
Comment (HMCTSDeveloper) [reply src/main/resources/application-stg.yaml:1]: Ye I think one of my merges brought it back in. ACCEPT

File: src/main/resources/db/migration/common/V1_218__node_register_missing_fk_and_seq.sql (hunk 2 above)
Comment (HMCTSDeveloper) [reply src/main/resources/db/migration/common/V1_218__node_register_missing_fk_and_seq.sql:6]: Going to keep these changes as has been discussed with Sashi and Barry. The new proposed changes in v50 CREATE TABLE node_register
(node_id                     INTEGER                       NOT NULL  --reusing PK from legacy
,ctr_id                      INTEGER                       NOT NULL
,node_type                   CHARACTER VARYING             NOT NULL   DEFAULT 'DAR'
,hostname                    CHARACTER VARYING             NOT NULL
,ip_address                  CHARACTER VARYING             NOT NULL
,mac_address                 CHARACTER VARYING             NOT NULL
,created_ts                  TIMESTAMP WITH TIME ZONE      NOT NULL
,created_by                  INTEGER                       NOT NULL
,last_modified_ts            TIMESTAMP WITH TIME ZONE      NOT NULL
,last_modified_by            INTEGER                       NOT NULL
) TABLESPACE darts_tables;

File: src/main/resources/db/migration/common/V1_217__drop_courthouse_region_cra_id.sql (hunk 3 above)
Comment (HMCTSDeveloper) [reply src/main/resources/db/migration/common/V1_217__drop_courthouse_region_cra_id.sql:5]: I have kept the changes in chronological order that's why they are split. this change was missed in v47 and the dropping of the sequence was in v48.

File: src/main/resources/db/migration/common/V1_218__node_register_missing_fk_and_seq.sql (hunk 4 above)
Comment (HMCTSDeveloper) [reply src/main/resources/db/migration/common/V1_218__node_register_missing_fk_and_seq.sql:4]: No we are going forward with node_register table.

File: src/main/resources/db/migration/common/V1_219__v48_without_transcription.sql (hunk 5 above)
Comment (HMCTSDeveloper) [reply src/main/resources/db/migration/common/V1_219__v48_without_transcription.sql:6]: This was done as part of v48 that's why is here.

**Review by davet1985 — APPROVED** [review]

**Pull request merged**